generators/pkg/generator/generate.go

```diff
@@ -23,6 +27,13 @@ type GitHubSpec struct {
 	Steps []v1beta1.Step `json:"steps,omitempty"`
 }
 
+// Trigger includes the Tekton trigger resources
+type Trigger struct {
```

Comment:
Let's avoid exposing new types that may cause confusion with existing trigger types. I think it's fine to use this for grouping, but let's unexport this so it can't be used by other packages.